### PR TITLE
Attach Filer to appshell before loading dependencies that require it

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

**Comment by humphd** [comment]
@K-LV can you test locally?  r=me with a +1 from Klever.

**Comment by gideonthomas** [comment]
@humphd does this look ok?
@K-LV can you test with this?

**Comment by gideonthomas** [comment]
sorry gonna make one last change

**Comment by gideonthomas** [comment]
done

File: src/filesystem/impls/filer/BracketsFiler.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, appshell */
```
Comment (humphd) [inline src/filesystem/impls/filer/BracketsFiler.js]: Remove `appshell` here.  Also, you should run `grunt jshint` to see if you are going to have other lint errors.

File: src/filesystem/impls/filer/BracketsFiler.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        if(!fs) {
+            fs = new Filer.FileSystem({provider: new Filer.FileSystem.providers.Memory()});
+        }
+
+        return fs;
+    };
+
+    fs = Filer.fs();
+
+    module.exports = appshell.Filer = Filer;
```
Comment (humphd) [inline src/filesystem/impls/filer/BracketsFiler.js]: Don't put this on `appshell`

**Comment by humphd** [comment]
r=me with fixes made.  This is probably going to require a follow-up patch in the `brackets-browser-livedev` to use the `BracketsFiler` module vs. `appshelll.Filer`.  Can you do that too?

File: src/filesystem/impls/filer/BracketsFiler.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    Filer.fs = function() {
+        if(!fs) {
+            fs = new Filer.FileSystem({provider: new Filer.FileSystem.providers.Memory()});
+        }
+
+        return fs;
+    };
+
+    fs = Filer.fs();
```
Comment (humphd) [inline src/filesystem/impls/filer/BracketsFiler.js]: You can remove this line, since lines 10-12 will do it on first call.

**Pull request merged**